internal/repository/file.go

```diff
@@ -0,0 +1,298 @@
+var _ repository.WritableRepository = (*FileRepository)(nil)
+var _ repository.HierarchicalRepository = (*FileRepository)(nil)
+var _ repository.ListableRepository = (*FileRepository)(nil)
+var _ repository.MovableRepository = (*FileRepository)(nil)
+var _ repository.CopyableRepository = (*FileRepository)(nil)
+
+var _ fyne.URIReadCloser = (*file)(nil)
+var _ fyne.URIWriteCloser = (*file)(nil)
+
+type file struct {
+	*os.File
+	uri fyne.URI
+}
+
+func (f *file) URI() fyne.URI {
+	return f.uri
+}
+
+// FileRepository implements a simple wrapper around golang's filesystem
+// interface libraries. It should be registered by the driver on platforms
+// where it is appropriate to do so.
+//
+// This repository is suitable to handle the file:// scheme.
+//
+// Since: 2.0.0
+type FileRepository struct {
+}
+
+// NewFileRepository creates a new FileRepository instance.
+// The caller needs to call repository.Register() with the result of this function.
+//
+// Since: 2.0.0
+func NewFileRepository() *FileRepository {
+	return &FileRepository{}
+}
+
+// Exists implements repository.Repository.Exists
+//
+// Since: 2.0.0
+func (r *FileRepository) Exists(u fyne.URI) (bool, error) {
+	p := u.Path()
+
+	_, err := os.Stat(p)
+	ok := false
+
+	if err == nil {
+		ok = true
+	} else if os.IsNotExist(err) {
+		err = nil
+	}
+
+	return ok, err
+}
+
+func openFile(uri fyne.URI, create bool) (*file, error) {
+	path := uri.Path()
+	var f *os.File
+	var err error
+	if create {
+		f, err = os.Create(path) // If it exists this will truncate which is what we wanted
+	} else {
+		f, err = os.Open(path)
+	}
+	return &file{File: f, uri: uri}, err
+}
+
+// Reader implements repository.Repository.Reader
+//
+// Since: 2.0.0
+func (r *FileRepository) Reader(u fyne.URI) (fyne.URIReadCloser, error) {
+	return openFile(u, false)
+}
+
+// CanRead implements repository.Repository.CanRead
+//
+// Since: 2.0.0
+func (r *FileRepository) CanRead(u fyne.URI) (bool, error) {
+	f, err := os.OpenFile(u.Path(), os.O_RDONLY, 0666)
+	if err == nil {
+		f.Close()
+	} else {
+
+		if os.IsPermission(err) {
+			return false, nil
+		}
+
+		if os.IsNotExist(err) {
+			return false, nil
+		}
+
+		return false, err
+	}
+
+	return true, nil
+}
+
+// Destroy implements repository.Repository.Destroy
+func (r *FileRepository) Destroy(scheme string) {
+	// do nothing
+}
+
+// Writer implements repository.WritableRepository.Writer
+//
+// Since: 2.0.0
+func (r *FileRepository) Writer(u fyne.URI) (fyne.URIWriteCloser, error) {
+	return openFile(u, true)
+}
+
+// CanWrite implements repository.WritableRepository.CanWrite
+//
+// Since: 2.0.0
+func (r *FileRepository) CanWrite(u fyne.URI) (bool, error) {
+	f, err := os.OpenFile(u.Path(), os.O_WRONLY, 0666)
+	if err == nil {
+		f.Close()
+	} else {
+
+		if os.IsPermission(err) {
+			return false, nil
+		}
+
+		if os.IsNotExist(err) {
+			// We may need to do extra logic to check if the
+			// directory is writeable, but presumably the
```

Comment:
```suggestion
			// directory is writable, but presumably the
```

Reply:
Oops, when I ran `sed` I only did the uppercase version...

Reply:
bafe9c8ae48d43f7c2537005d6e1498e65adbeff